util/src/testing.rs

```diff
@@ -0,0 +1,95 @@
+use core::fmt;
+use core::mem;
+use core::slice;
+
+/// Internal definition type used for a test.
```

Comment:
nit: the test runner in the kernel consumes this, so i am not sure if i would call it "internal" :P